common/src/main/scala/tsec/common/ByteUtils.scala

```diff
@@ -4,15 +4,54 @@ import java.security.MessageDigest
 
 object ByteUtils {
 
-  @inline def zeroByteArray(a: Array[Byte]): Unit = {
+  @inline def zeroByteArray(a: Array[Byte]): Unit =
     java.util.Arrays.fill(a, 0.toByte)
-  }
 
-  @inline def zeroCharArray(a: Array[Char]): Unit = {
+  @inline def zeroCharArray(a: Array[Char]): Unit =
     java.util.Arrays.fill(a, 0.toChar)
-  }
 
   @inline def constantTimeEquals(a: Array[Byte], b: Array[Byte]): Boolean =
     MessageDigest.isEqual(a, b)
 
+  @inline def intToBytes(a: Int): Array[Byte] =
+    Array(
+      (a >> 24).toByte,
+      (a >> 16).toByte,
+      (a >> 8).toByte,
+      a.toByte
+    )
+
+  @inline def unsafeBytesToInt(a: Array[Byte]): Int =
+    a(0) << 24 | (a(1) & 0xFF) << 16 | (a(2) & 0xFF) << 8 | (a(3) & 0xFF)
+
+  @inline def bytesToInt(a: Array[Byte]): Option[Int] =
+    if (a.length < Integer.BYTES) None
```

Comment:
shouldn't this `a.length != Integer.BYTES`

Reply:
good question: Do we want to fail all byte arrays that are not the exact size, or should we be lenient on arrays that are at least N bytes?

Reply:
In my opinion we should fail if the size doesn't match.